rewrite-xml/src/main/java/org/openrewrite/xml/SpringJavaTypeReferenceProvider.java

```diff
@@ -0,0 +1,51 @@
+/*
+ * Copyright 2024 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.openrewrite.xml;
+
+import org.openrewrite.JavaTypeReferenceProvider;
+import org.openrewrite.SourceFile;
+import org.openrewrite.trait.JavaTypeReference;
+import org.openrewrite.xml.trait.SpringJavaTypeReference;
+import org.openrewrite.xml.tree.Xml;
+
+import java.util.HashSet;
+import java.util.Set;
+
+public class SpringJavaTypeReferenceProvider implements JavaTypeReferenceProvider {
```

Comment:
AFAIK the service implementation classes are allowed to be package private.

```suggestion
class SpringJavaTypeReferenceProvider implements JavaTypeReferenceProvider {
```

Reply:
Seems the implementation has to be public: 
```Caused by: java.lang.IllegalAccessException: class java.util.ServiceLoader$ProviderImpl (in module java.base) cannot access a member of class org.openrewrite.xml.trait.SpringTypeReferenceProvider with modifiers "public"```